packages/scene-composer/src/augmentations/components/three-fiber/anchor/AnchorWidget.tsx

```diff
@@ -124,11 +128,48 @@ export function AsyncLoadedAnchorWidget({
       WarningIconSvgString,
       ErrorIconSvgString,
       VideoIconSvgString,
+      CustomIconSvgString,
     ];
     return iconStrings.map((iconString, index) => {
+      if (keys[index] === 'Custom') {
+        const parser = new DOMParser();
+        const svgDocument = parser.parseFromString(iconString, 'image/svg+xml');
+        const svgRoot = svgDocument.documentElement;
+
+        const replaceFillAttribute = (element: Element) => {
+          const tagName = element.tagName.toLowerCase();
+          if (tagName === 'ellipse') {
+            if (chosenColor === undefined) {
+              return;
+            } else {
+              THREE.Cache.clear();
```

Comment:
It seems like doing a Three.Cache.remove instead of clear would be better if possible.

Reply:
Actually it looks like this and the traverse function could use the new util library function you wrote instead.

Reply:
Is there a way to use the util functions you wrote instead of having these custom versions in this class?

Reply:
Yeah. I can see if that can be optimized.